src/llama-model-loader.cpp

```diff
@@ -1064,9 +1075,14 @@ bool llama_model_loader::load_all_data(
             }
         } else {
             const auto & file = files.at(weight->idx);
+
             if (ggml_backend_buffer_is_host(cur->buffer)) {
-                file->seek(weight->offs, SEEK_SET);
-                file->read_raw(cur->data, n_size);
+                if (file->has_direct_io()) {
+                    file->read_aligned_chunk(weight->offs, cur->data, n_size, alignment);
+                } else {
+                    file->seek(weight->offs, SEEK_SET);
+                    file->read_raw(cur->data, n_size);
+                }
```

Comment:
nit: can collapse these branches using a new `llama_file::read_raw_at()`:

```c++
file->read_raw_at(cur->data, n_size, weight->offs);
```

Internally the implementation will do alignment if direct io is available. In that case `read_aligned_chunk()` is not needed.